x/auth/client/cli/sign.go

```diff
@@ -77,17 +105,31 @@ func makeSignCmd(cdc *amino.Codec, decoder auth.AccountDecoder) func(cmd *cobra.
 	}
 }
 
-func printSignatures(stdTx auth.StdTx) {
+func printSignatures(stdTx auth.StdTx) bool {
 	fmt.Println("Signers:")
-	for i, signer := range stdTx.GetSigners() {
+	signers := stdTx.GetSigners()
+	for i, signer := range signers {
 		fmt.Printf(" %v: %v\n", i, signer.String())
 	}
+
+	sigs := stdTx.GetSignatures()
 	fmt.Println("")
 	fmt.Println("Signatures:")
+	success := true
+	if len(sigs) != len(signers) {
+		success = false
+	}
 	for i, sig := range stdTx.GetSignatures() {
-		fmt.Printf(" %v: %v\n", i, sdk.AccAddress(sig.Address()).String())
+		sigAddr := sdk.AccAddress(sig.Address())
+		sigSanity := "OK"
+		if i >= len(signers) || !sigAddr.Equals(signers[i]) {
```

Comment:
Huh maybe I missed it but where in this logic do we actually check that the signatures are valid? Are we not intending to do that (perhaps because we don't have the public keys)? If not we need to make that *very* clear, users will assume a command called `validate-signatures` actually validates the signatures.

Reply:
We don't check the signatures. I'll make it clear in the flag's docs